File: Teste_banco/Banco.py
import locale

locale.setlocale(locale.LC_ALL, 'pt_BR.UTF-8')  # Para formatar como moeda brasileira

class Banco:
    def __init__(self, nome, numero, agencia):
        # Atributos "protegidos" (por convenção em Python, usamos underline)
        self._nome = nome
        self._numero = numero
        self._agencia = agencia
        self._saldo = 0.0

    # Método para depositar valor
    def depositar(self, valor):
        self._saldo += valor

    # Método para sacar valor
    def sacar(self, valor):
        if self._saldo >= valor:
            self._saldo -= valor
        else:
            print("Saldo insuficiente")

    # Método para mostrar os dados da conta
    def mostrar(self):
        print(f"Nome: {self._nome}")
        print(f"Número: {self._numero}")
        print(f"Agência: {self._agencia}")
        print("Saldo: R$", locale.currency(self._saldo, grouping=True))


# Programa principal
if __name__ == "__main__":
    nova_conta = Banco("Sarah", 1, 1)

    nova_conta.depositar(100.0)
    nova_conta.mostrar()

    nova_conta.sacar(20.5)
    nova_conta.mostrar()

File: Teste_banco/Banco.java
// Aqui temos encapsulamento, que é basicamente os atributos do objeto de forma protegida.
package teste_POO;
import java.text.DecimalFormat;

//Criação da classe e suas variaveis
public class Banco {
	private int numero; 
	private int agencia;
	private String nome;
	private double saldo;
	DecimalFormat dfDinheiro = new DecimalFormat("###,##0.00");

// Construtor que inicializa os atributos da classe, mesmo que estejam encapsulados (privados),
//porque estamos dentro da própria classe.
public Banco (String nome, int numero, int agencia) {
	this.nome = nome; // O this faz referencia ao parametro do Objeto.
	this.numero = numero;
	this.agencia = agencia;            
	this.saldo = 0.00;
} 

//Função para depositar algum valor
void depositar(Double valor) {
	this.saldo += valor;
}

//Função para sacar algum valor
void sacar(Double valor) {
	if (this.saldo >= valor) {
		this.saldo -= valor;
	} else { 
		System.out.println("Saldo insuficiente");
	}
}

// Função para mostrar os dados
void mostrar() {
System.out.println("Nome: " + this.nome);
System.out.println("Número: " + this.numero);
System.out.println("Agência: " + this.agencia);
System.out.println("Saldo: R$ " + dfDinheiro.format(this.saldo));}

}                
